src/experiment_prototype/experiment_slice.py

```diff
@@ -587,19 +591,21 @@ def check_freq(cls, freq):
 
     @validator('xcf', always=True)
     def check_xcf(cls, xcf, values):
```

Comment:
I feel like this function might be missing a `return xcf` at the end after the `raise ValueError` statement.